tests/api_spec.rs

```diff
@@ -0,0 +1,145 @@
+#[cfg(test)]
+mod test {
+  use std::collections::BTreeMap;
+  use std::fs;
+  use std::str::FromStr;
+  use std::sync::Arc;
+
+  use async_graphql::InputType;
+  use http_cache_semantics::RequestLike;
+  use hyper::{Body, Request};
+  use reqwest::header::{HeaderName, HeaderValue};
+  use serde::Deserialize;
+  use tailcall::blueprint::Blueprint;
+  use tailcall::config::Config;
+  use tailcall::http::{graphql_request, HttpClient, Method, Response, ServerContext};
+  use url::Url;
+
+  #[derive(Deserialize, Clone, Debug)]
+  pub struct APIRequest {
+    method: Method,
+    pub url: Url,
+    pub headers: Option<BTreeMap<String, String>>,
+    pub body: Option<serde_json::Value>,
+  }
+  #[derive(Deserialize, Clone, Debug)]
+  pub struct APIResponse {
+    pub status: u16,
+    pub headers: Option<BTreeMap<String, String>>,
+    pub body: Option<serde_json::Value>,
+  }
+  #[derive(Deserialize, Clone, Debug)]
+  pub struct UpstreamRequest(pub APIRequest);
+  #[derive(Deserialize, Clone, Debug)]
+  pub struct UpstreamResponse(APIResponse);
+  #[derive(Deserialize)]
+  pub struct DownstreamRequest(pub APIRequest);
+  #[derive(Deserialize)]
+  pub struct DownstreamResponse(pub APIResponse);
+  #[derive(Deserialize)]
+  pub struct DownstreamAssertion {
+    pub request: DownstreamRequest,
+    pub response: DownstreamResponse,
+  }
+
+  #[derive(Default, Deserialize)]
+  pub struct APISpecification {
+    pub config: String,
+    pub name: String,
+    pub description: Option<String>,
+    pub upstream_mocks: Vec<(UpstreamRequest, UpstreamResponse)>,
+    pub expected_upstream_requests: Vec<UpstreamRequest>,
+    pub downstream_assertions: Vec<DownstreamAssertion>,
+  }
```

Comment:
Take inspiration from https://github.com/tailcallhq/tailcall/issues/603
To allow developers to control which spec to run/ignore or expect to fail.